server/sonar-db-dao/src/main/java/org/sonar/db/rule/RuleMetadataDto.java

```diff
@@ -0,0 +1,188 @@
+
+public class RuleMetadataDto {
+  private int ruleId;
+  private String organizationUuid;
+  private String noteData;
+  private String noteUserLogin;
+  private Long noteCreatedAt;
+  private Long noteUpdatedAt;
+  private String remediationFunction;
+  private String remediationGapMultiplier;
+  private String remediationBaseEffort;
+  private String tags;
+  private long createdAt;
+  private long updatedAt;
+
+  public int getRuleId() {
+    return ruleId;
+  }
+
+  public RuleMetadataDto setRuleId(int ruleId) {
+    this.ruleId = ruleId;
+    return this;
+  }
+
+  public String getOrganizationUuid() {
+    return organizationUuid;
+  }
+
+  public RuleMetadataDto setOrganizationUuid(String organizationUuid) {
+    this.organizationUuid = organizationUuid;
+    return this;
+  }
+
+  @CheckForNull
+  public String getNoteData() {
+    return noteData;
+  }
+
+  public RuleMetadataDto setNoteData(@Nullable String s) {
+    this.noteData = s;
+    return this;
+  }
+
+  @CheckForNull
+  public String getNoteUserLogin() {
+    return noteUserLogin;
+  }
+
+  public RuleMetadataDto setNoteUserLogin(@Nullable String noteUserLogin) {
+    this.noteUserLogin = noteUserLogin;
+    return this;
+  }
+
+  @CheckForNull
+  public Long getNoteCreatedAt() {
+    return noteCreatedAt;
+  }
+
+  public RuleMetadataDto setNoteCreatedAt(@Nullable Long noteCreatedAt) {
+    this.noteCreatedAt = noteCreatedAt;
+    return this;
+  }
+
+  @CheckForNull
+  public Long getNoteUpdatedAt() {
+    return noteUpdatedAt;
+  }
+
+  public RuleMetadataDto setNoteUpdatedAt(@Nullable Long noteUpdatedAt) {
+    this.noteUpdatedAt = noteUpdatedAt;
+    return this;
+  }
+
+  @CheckForNull
+  public String getRemediationFunction() {
+    return remediationFunction;
+  }
+
+  public RuleMetadataDto setRemediationFunction(@Nullable String remediationFunction) {
+    this.remediationFunction = remediationFunction;
+    return this;
+  }
+
+  @CheckForNull
+  public String getRemediationGapMultiplier() {
+    return remediationGapMultiplier;
+  }
+
+  public RuleMetadataDto setRemediationGapMultiplier(@Nullable String remediationGapMultiplier) {
+    this.remediationGapMultiplier = remediationGapMultiplier;
+    return this;
+  }
+
+  @CheckForNull
+  public String getRemediationBaseEffort() {
+    return remediationBaseEffort;
+  }
+
+  public RuleMetadataDto setRemediationBaseEffort(@Nullable String remediationBaseEffort) {
+    this.remediationBaseEffort = remediationBaseEffort;
+    return this;
+  }
+
+  public Set<String> getTags() {
+    return tags == null ? new HashSet<>() : new TreeSet<>(Arrays.asList(StringUtils.split(tags, ',')));
+  }
+
+  public RuleMetadataDto setTags(Set<String> tags) {
+    String raw = tags.isEmpty() ? null : StringUtils.join(tags, ',');
+    checkArgument(raw == null || raw.length() <= 4000, "Rule tags are too long: %s", raw);
+    this.tags = raw;
+    return this;
+  }
+
+  private String getTagsField() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "getTagsField" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)